src/ui/components/SelectedFrameContextAdapter.tsx

```diff
@@ -0,0 +1,11 @@
+import { SelectedFrameContext } from "bvaughn-architecture-demo/src/contexts/SelectedFrameContext";
+import { getSelectedFrameId } from "devtools/client/debugger/src/selectors";
+import { PropsWithChildren, useContext } from "react";
+import { useAppSelector } from "ui/setup/hooks";
+
+export default function SelectedFrameContextAdapter({ children }: PropsWithChildren) {
+  const selectedFrameId = useAppSelector(getSelectedFrameId);
+  const { setSelectedFrameId } = useContext(SelectedFrameContext);
+  setSelectedFrameId(selectedFrameId);
```

Comment:
React is bailing out on equality for us here, to avoid an infinite loop, but it would be nice if we conditionally called this instead, just to be clearer:

```suggestion
  const selectedFrameIdRedux = useAppSelector(getSelectedFrameId);
  const { selectedFrameId, setSelectedFrameId } = useContext(SelectedFrameContext);
  if (selectedFrameId !== selectedFrameIdRedux) {
    setSelectedFrameId(selectedFrameId);
  }
```